disnake/ext/commands/cog.py

```diff
@@ -240,10 +268,24 @@ def __new__(cls: Type[CogT], *args: Any, **kwargs: Any) -> CogT:
         # To do this, we need to interfere with the Cog creation process.
         self = super().__new__(cls)
         cmd_attrs = cls.__cog_settings__
+        slash_cmd_attrs = cls.__cog_slash_settings__
+        user_cmd_attrs = cls.__cog_user_settings__
+        message_cmd_attrs = cls.__cog_message_settings__
 
         # Either update the command with the cog provided defaults or copy it.
         # r.e type ignore, type-checker complains about overriding a ClassVar
         self.__cog_commands__ = tuple(c._update_copy(cmd_attrs) for c in cls.__cog_commands__)  # type: ignore
+        cog_app_commands: List[InvokableApplicationCommand] = []
+        for c in cls.__cog_app_commands__:
+            if isinstance(c, (InvokableSlashCommand, SubCommandGroup, SubCommand)):
+                updated = c._update_copy(slash_cmd_attrs)
+            elif isinstance(c, InvokableUserCommand):
+                updated = c._update_copy(user_cmd_attrs)
+            else:
```

Comment:
this should check the command is an instance of InvokableMessageCommand, rather than assume (even though it should be)